<a href="https://colab.research.google.com/github/etcex2969-spec/-AIFFEL_quest_eng/blob/main/Security_Performance_Visualization'%EC%82%AC%EB%B3%B8%EC%9D%98_%EC%82%AC%EB%B3%B8.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [18]:
!pip install python-pptx

In [19]:
import os
from pptx import Presentation

file_path = '/content/Autonomous Financial Security Center.pptx'

# 파일 존재 여부 확인
if os.path.exists(file_path):
    prs = Presentation(file_path)

    print(f"Total Slides: {len(prs.slides)}\n")

    for i, slide in enumerate(prs.slides):
        print(f"--- Slide {i+1} ---")
        for shape in slide.shapes:
            if hasattr(shape, "text"):
                print(shape.text)
        print("\n")
else:
    print(f"오류: '{file_path}' 파일을 찾을 수 없습니다.")
    print("Colab 왼쪽 메뉴의 '파일' 탭을 통해 해당 PPTX 파일을 업로드한 후 다시 실행해 주세요.")

오류: '/content/Autonomous Financial Security Center.pptx' 파일을 찾을 수 없습니다.
Colab 왼쪽 메뉴의 '파일' 탭을 통해 해당 PPTX 파일을 업로드한 후 다시 실행해 주세요.


In [20]:
from google.colab import drive
# force_remount=True를 추가하여 기존 연결 오류를 초기화하고 다시 시도합니다.
drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


In [21]:
from pptx import Presentation
from pptx.util import Inches, Pt

try:
    file_path = '/content/Autonomous Financial Security Center.pptx'
    prs = Presentation(file_path)

    # Slide 11: Fill in Collaborative Agent Roles
    # 슬라이드 인덱스는 0부터 시작하므로 11번째 슬라이드는 index 10입니다.
    slide11 = prs.slides[10]

    # 기존 텍스트 상자와 겹치지 않도록 적절한 위치(Inches(1), Inches(1.5))에 추가
    content_box = slide11.shapes.add_textbox(Inches(1), Inches(1.5), Inches(8), Inches(4))
    tf = content_box.text_frame
    tf.word_wrap = True

    # 에이전트 역할 추가
    tf.text = "• Security Analyst Agent: Real-time threat detection and log analysis."
    p = tf.add_paragraph()
    p.text = "• Critic Agent: Audits analyst decisions and validates against historical data."
    p = tf.add_paragraph()
    p.text = "• Memory Manager: Indexes successful defense patterns into the Vector DB."
    p = tf.add_paragraph()
    p.text = "• Response Agent: Executes automated containment protocols (v2.0)."

    # Slide 2: 문구 개선
    slide2 = prs.slides[1]
    for shape in slide2.shapes:
        if hasattr(shape, "text") and "How AI agents learn" in shape.text:
            shape.text = "Autonomous learning through iterative feedback loops, transforming raw threat data into actionable security intelligence without manual tuning."

    output_path = '/content/Autonomous_Financial_Security_Center_Improved.pptx'
    prs.save(output_path)
    print(f"Successfully saved improved file to: {output_path}")

except Exception as e:
    print(f"Error: {e}")
    print("파일이 /content/ 경로에 업로드되었는지 다시 한번 확인해 주세요.")

Error: Package not found at '/content/Autonomous Financial Security Center.pptx'
파일이 /content/ 경로에 업로드되었는지 다시 한번 확인해 주세요.


In [22]:
import os
print("Files in /content/:")
print(os.listdir('/content/'))

Files in /content/:
['.config', 'drive', 'Autonomous_Financial_Security_Center_Final_v2.pptx', 'Autonomous_Financial_Security_Center_Improved.pptx', 'Autonomous_Financial_Security_Center_Final_V1.pptx', 'sample_data']


In [23]:
import os
print("Files in /content/:")
print(os.listdir('/content/'))

Files in /content/:
['.config', 'drive', 'Autonomous_Financial_Security_Center_Final_v2.pptx', 'Autonomous_Financial_Security_Center_Improved.pptx', 'Autonomous_Financial_Security_Center_Final_V1.pptx', 'sample_data']


In [24]:
from pptx import Presentation
from pptx.util import Inches
import os

# 현재 존재하는 파일명으로 경로를 수정했습니다.
file_path = '/content/Autonomous_Financial_Security_Center_Final_v2.pptx'

if os.path.exists(file_path):
    prs = Presentation(file_path)

    # 1. Slide 2 문구 업데이트
    if len(prs.slides) > 1:
        slide2 = prs.slides[1]
        for shape in slide2.shapes:
            if hasattr(shape, 'text') and "How AI agents learn" in shape.text:
                shape.text = "Autonomous learning through iterative feedback loops, transforming raw threat data into actionable security intelligence without manual tuning."

    # 2. Slide 11 에이전트 역할 추가
    if len(prs.slides) > 10:
        slide11 = prs.slides[10]
        content_box = slide11.shapes.add_textbox(Inches(1), Inches(1.5), Inches(8), Inches(4))
        tf = content_box.text_frame
        tf.word_wrap = True
        tf.text = "• Security Analyst Agent: Real-time threat detection and log analysis."
        p2 = tf.add_paragraph(); p2.text = "• Critic Agent: Audits analyst decisions and validates against historical data."
        p3 = tf.add_paragraph(); p3.text = "• Memory Manager: Indexes successful defense patterns into the Vector DB."
        p4 = tf.add_paragraph(); p4.text = "• Response Agent: Executes automated containment protocols (v2.0)."

    # 3. Slide 12 결론 문구 강화
    if len(prs.slides) > 11:
        slide12 = prs.slides[11]
        for shape in slide12.shapes:
            if hasattr(shape, 'text') and "Evolution Never Sleeps" in shape.text:
                shape.text = "Evolution Never Sleeps: Secure Your Financial Future with Autonomous AI."

    output_path = '/content/Autonomous_Financial_Security_Center_Improved.pptx'
    prs.save(output_path)
    print(f"Successfully updated and saved to: {output_path}")
else:
    print(f"오류: {file_path} 파일을 찾을 수 없습니다. 파일명을 확인해 주세요.")

Successfully updated and saved to: /content/Autonomous_Financial_Security_Center_Improved.pptx


In [25]:
from pptx import Presentation
from pptx.util import Inches, Pt
import os

# 설명: PackageNotFoundError는 지정된 경로에 파일이 없을 때 발생합니다.
file_path = '/content/Autonomous Financial Security Center.pptx'

if os.path.exists(file_path):
    prs = Presentation(file_path)

    # Slide 11: Collaborative Agent Roles 보완
    slide11 = prs.slides[10]
    # 텍스트가 겹치지 않도록 적절한 위치에 텍스트 박스 추가
    content_box = slide11.shapes.add_textbox(Inches(1), Inches(1.5), Inches(8), Inches(4))
    tf = content_box.text_frame
    tf.word_wrap = True
    tf.text = "• Security Analyst Agent: Real-time threat detection and log analysis."
    p = tf.add_paragraph()
    p.text = "• Critic Agent: Audits analyst decisions and validates against historical data."
    p = tf.add_paragraph()
    p.text = "• Memory Manager: Indexes successful defense patterns into the Vector DB."
    p = tf.add_paragraph()
    p.text = "• Response Agent: Executes automated containment protocols (v2.0)."

    # Slide 2: 학습 메커니즘 문구 개선
    slide2 = prs.slides[1]
    for shape in slide2.shapes:
        if hasattr(shape, "text") and "How AI agents learn" in shape.text:
            shape.text = "Autonomous learning through iterative feedback loops, transforming raw threat data into actionable security intelligence without manual tuning."

    output_path = '/content/Autonomous_Financial_Security_Center_Improved.pptx'
    prs.save(output_path)
    print(f"성공적으로 수정되어 다음 경로에 저장되었습니다: {output_path}")
else:
    print(f"오류: {file_path} 파일을 찾을 수 없습니다. 파일을 업로드한 후 다시 시도해 주세요.")

오류: /content/Autonomous Financial Security Center.pptx 파일을 찾을 수 없습니다. 파일을 업로드한 후 다시 시도해 주세요.


In [26]:
from pptx import Presentation
from pptx.util import Inches, Pt

# 기존에 생성된 개선된 파일을 다시 엽니다.
file_path = '/content/Autonomous_Financial_Security_Center_Improved.pptx'
prs = Presentation(file_path)

# 목차/요약 슬라이드를 추가합니다 (제목 및 내용 레이아웃 사용)
# 보통 인덱스 1은 'Title and Content' 레이아웃인 경우가 많습니다.
blank_slide_layout = prs.slide_layouts[1]
slide = prs.slides.add_slide(blank_slide_layout)

# 슬라이드를 두 번째 위치(index 1)로 이동시키는 기능은 python-pptx에서 직접 지원하지 않으므로
# 새 슬라이드에 내용을 채운 뒤, 나중에 수동으로 순서를 조정하거나 내용을 구성합니다.
# 여기서는 일단 마지막에 추가하여 내용을 제안합니다.

title = slide.shapes.title
title.text = "Executive Summary"

tf = slide.placeholders[1].text_frame
tf.text = "Overview of Autonomous Financial Security"

points = [
    "Integrated AI Defense: Synergizing LLMs with Security Operations.",
    "Collaborative Intelligence: Analyst, Critic, and Memory Agents working in tandem.",
    "Proactive Containment: Reducing MTTR (Mean Time To Respond) via automated protocols.",
    "Continuous Evolution: Feedback-driven learning for future-proof security."
]

for point in points:
    p = tf.add_paragraph()
    p.text = point
    p.level = 1

output_path = '/content/Autonomous_Financial_Security_Center_Final_V1.pptx'
prs.save(output_path)
print(f"요약 슬라이드가 추가된 최종 파일이 저장되었습니다: {output_path}")

요약 슬라이드가 추가된 최종 파일이 저장되었습니다: /content/Autonomous_Financial_Security_Center_Final_V1.pptx


In [27]:
from pptx import Presentation
from pptx.chart.data import CategoryChartData
from pptx.enum.chart import XL_CHART_TYPE
from pptx.util import Inches

# 기존 파일 로드
file_path = '/content/Autonomous_Financial_Security_Center_Final_V1.pptx'
prs = Presentation(file_path)

# --- 1. 이미지 추가 예시 (새 슬라이드에 추가) ---
slide_image = prs.slides.add_slide(prs.slide_layouts[1])
slide_image.shapes.title.text = "Visualizing Security Data (Image Example)"

# 참고: /content/sample_data/ 아래의 이미지가 없으므로 실제 파일 경로를 넣어야 합니다.
# 여기서는 코드 예시만 보여드립니다.
# left, top, width = Inches(1), Inches(2), Inches(6)
# slide_image.shapes.add_picture('image_path.png', left, top, width)

# --- 2. 차트 추가 예시 (새 슬라이드에 추가) ---
slide_chart = prs.slides.add_slide(prs.slide_layouts[1])
slide_chart.shapes.title.text = "Threat Detection Metrics"

chart_data = CategoryChartData()
chart_data.categories = ['Phishing', 'Malware', 'DDOS', 'SQL Injection']
chart_data.add_series('Detected Threats', (120, 85, 40, 25))

x, y, cx, cy = Inches(1), Inches(2), Inches(8), Inches(4)
slide_chart.shapes.add_chart(
    XL_CHART_TYPE.COLUMN_CLUSTERED, x, y, cx, cy, chart_data
)

# 저장
final_output = '/content/Autonomous_Financial_Security_Center_Final_V2.pptx'
prs.save(final_output)
print(f"이미지 및 차트 예시가 추가된 파일이 저장되었습니다: {final_output}")

이미지 및 차트 예시가 추가된 파일이 저장되었습니다: /content/Autonomous_Financial_Security_Center_Final_V2.pptx


In [28]:
from pptx import Presentation
from pptx.chart.data import CategoryChartData
from pptx.enum.chart import XL_CHART_TYPE
from pptx.util import Inches

# 파일 로드 (이전 단계에서 생성된 V2 활용)
file_path = '/content/Autonomous_Financial_Security_Center_Final_V2.pptx'
prs = Presentation(file_path)

# 신규 차트 데이터를 정의합니다 (예시 보안 위협 데이터)
categories = ['인증 시도 실패', '악성 코드 탐지', '네트워크 스캔', '비정상 접속']
series_data = [450, 120, 310, 85]

# 새로운 슬라이드 추가
slide = prs.slides.add_slide(prs.slide_layouts[1])
slide.shapes.title.text = "Real-time Security Threat Statistics"

# 차트 데이터 구성
chart_data = CategoryChartData()
chart_data.categories = categories
chart_data.add_series('Detection Count', series_data)

# 차트 삽입 위치 및 크기 설정
x, y, cx, cy = Inches(1), Inches(2), Inches(8), Inches(4.5)
chart = slide.shapes.add_chart(
    XL_CHART_TYPE.COLUMN_CLUSTERED, x, y, cx, cy, chart_data
).chart

# 최종 파일 저장
final_v3_path = '/content/Autonomous_Financial_Security_Center_Final_V3.pptx'
prs.save(final_v3_path)
print(f"맞춤형 데이터가 반영된 최종 파일이 저장되었습니다: {final_v3_path}")

맞춤형 데이터가 반영된 최종 파일이 저장되었습니다: /content/Autonomous_Financial_Security_Center_Final_V3.pptx


In [29]:
from pptx import Presentation
from pptx.util import Pt
from pptx.dml.color import RGBColor

# 최종 버전 파일 로드
file_path = '/content/Autonomous_Financial_Security_Center_Final_V3.pptx'
prs = Presentation(file_path)

# 예시: 첫 번째 슬라이드(제목 슬라이드)의 폰트 수정
slide = prs.slides[0]
for shape in slide.shapes:
    if hasattr(shape, "text_frame"):
        for paragraph in shape.text_frame.paragraphs:
            for run in paragraph.runs:
                run.font.name = 'Arial'        # 폰트 이름 변경
                run.font.size = Pt(40)         # 폰트 크기 변경
                run.font.bold = True           # 굵게 설정
                run.font.color.rgb = RGBColor(0, 51, 102) # 진한 파란색으로 색상 변경

# 수정 사항 저장
style_updated_path = '/content/Autonomous_Financial_Security_Center_Styled.pptx'
prs.save(style_updated_path)
print(f"폰트 및 색상이 스타일링된 파일이 저장되었습니다: {style_updated_path}")

폰트 및 색상이 스타일링된 파일이 저장되었습니다: /content/Autonomous_Financial_Security_Center_Styled.pptx


In [30]:
from pptx import Presentation
from pptx.util import Pt
from pptx.dml.color import RGBColor

# 파일 로드
file_path = '/content/Autonomous_Financial_Security_Center_Styled.pptx'
prs = Presentation(file_path)

# 스타일 설정 값
THEME_FONT = 'Arial'
THEME_COLOR = RGBColor(0, 51, 102)  # Dark Blue

for slide in prs.slides:
    for shape in slide.shapes:
        if not hasattr(shape, "text_frame"):
            continue

        for paragraph in shape.text_frame.paragraphs:
            for run in paragraph.runs:
                run.font.name = THEME_FONT
                run.font.color.rgb = THEME_COLOR

                # 제목(Title) 타입의 도형인 경우 더 크게 설정
                if shape == slide.shapes.title:
                    run.font.size = Pt(40)
                    run.font.bold = True
                else:
                    run.font.size = Pt(18)

# 전체 테마가 적용된 파일 저장
full_theme_path = '/content/Autonomous_Financial_Security_Center_FullTheme.pptx'
prs.save(full_theme_path)
print(f"전체 슬라이드에 테마가 적용되었습니다: {full_theme_path}")

전체 슬라이드에 테마가 적용되었습니다: /content/Autonomous_Financial_Security_Center_FullTheme.pptx


In [31]:
from pptx import Presentation
from pptx.util import Inches
import os

# 파일 로드 (가장 최근 테마가 적용된 버전)
file_path = '/content/Autonomous_Financial_Security_Center_FullTheme.pptx'
prs = Presentation(file_path)

# 1. 삽입할 이미지 경로 설정 (업로드한 파일명으로 수정하세요)
image_path = '/content/your_logo_or_image.png' # 예: 'logo.png'

if os.path.exists(image_path):
    # 2. 이미지를 삽입할 슬라이드 선택 (예: 첫 번째 슬라이드)
    slide = prs.slides[0]

    # 3. 위치 및 크기 설정 (단위: 인치)
    left = Inches(0.5)
    top = Inches(0.5)
    width = Inches(1.5)  # 너비를 지정하면 비율에 맞춰 높이가 자동 조절됩니다.

    # 이미지 추가
    slide.shapes.add_picture(image_path, left, top, width=width)

    # 저장
    final_with_image = '/content/Autonomous_Financial_Security_Center_Final_Final.pptx'
    prs.save(final_with_image)
    print(f"이미지가 추가된 파일이 저장되었습니다: {final_with_image}")
else:
    print(f"오류: {image_path} 파일을 찾을 수 없습니다. 이미지를 업로드해 주세요.")

오류: /content/your_logo_or_image.png 파일을 찾을 수 없습니다. 이미지를 업로드해 주세요.


In [32]:
from pptx import Presentation

# 최신 파일 로드
file_path = '/content/Autonomous_Financial_Security_Center_FullTheme.pptx'
prs = Presentation(file_path)

print(f"=== 프레젠테이션 내용 요약 ({len(prs.slides)}개 슬라이드) ===\n")

for i, slide in enumerate(prs.slides):
    print(f"[Slide {i+1}]")
    text_content = []
    for shape in slide.shapes:
        if hasattr(shape, 'text') and shape.text.strip():
            text_content.append(shape.text.strip())

    if text_content:
        for line in text_content:
            print(f" - {line}")
    else:
        print(" (텍스트 없음 또는 이미지/차트 전용 슬라이드)")
    print("-" * 30)

=== 프레젠테이션 내용 요약 (20개 슬라이드) ===

[Slide 1]
 - Autonomous FinancialSecurity Center
 - Next-Generation Evolving AI Agent Workflows (v2.0)
------------------------------
[Slide 2]
 - The Mechanism ofSelf-Evolution
 - Autonomous learning through iterative feedback loops, transforming raw threat data into actionable security intelligence without manual tuning.
------------------------------
[Slide 3]
 - Legacy Systems
 - Evolving Engine v2
 - •
 - Fixed rule-based detection
 - •
 - Manual updates for new threats
 - •
 - High rate of false positives over time
 - •
 - Zero learning from past cycles
 - •
 - Dynamic agentic workflows
 - •
 - Self-reflection and critic-led feedback
 - •
 - Automatic insight accumulation in Memory
 - •
 - Decreasing error rates via self-correction
 - Static Automation vs. Evolving Intelligence
------------------------------
[Slide 4]
 - Self-Reflection Cycle
 - Every analysis triggers a Critic Agent. This agent audits the analyst's output, identifies missing da

In [33]:
from pptx import Presentation

# 최신 테마가 적용된 파일 로드
file_path = '/content/Autonomous_Financial_Security_Center_FullTheme.pptx'
prs = Presentation(file_path)

# 수정할 내용 정의 (슬라이드 인덱스: {기존 텍스트: 새 텍스트})
# 예: 4번째 슬라이드(index 3)의 내용을 더 구체화
slide_index = 3
slide = prs.slides[slide_index]

for shape in slide.shapes:
    if hasattr(shape, "text"):
        if "Every analysis triggers a Critic Agent" in shape.text:
            shape.text = "Each automated investigation initiates a Critic Agent audit, validating logic against historical threat vectors to ensure zero-day resilience."
        elif "In the next cycle" in shape.text:
            shape.text = "Retrieved memory embeddings empower the Analyst Agent to preemptively neutralize evolving attack patterns in subsequent cycles."

# 수정본 저장
refined_path = '/content/Autonomous_Financial_Security_Center_Refined.pptx'
prs.save(refined_path)
print(f"슬라이드 내용이 정교화되었습니다: {refined_path}")

슬라이드 내용이 정교화되었습니다: /content/Autonomous_Financial_Security_Center_Refined.pptx


In [34]:
from google.colab import files

# Final version path
final_file = '/content/Autonomous_Financial_Security_Center_Refined.pptx'

# Download the file to your local machine
files.download(final_file)
print(f"Downloading {final_file}...")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [35]:
from google.colab import files

# Download the latest refined version
files.download('/content/Autonomous_Financial_Security_Center_Refined.pptx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [36]:
from pptx import Presentation
import pandas as pd

# 파일 로드
file_path = '/content/Autonomous_Financial_Security_Center_Refined.pptx'
prs = Presentation(file_path)

slide_data = []

for i, slide in enumerate(prs.slides):
    title = ""
    if slide.shapes.title:
        title = slide.shapes.title.text
    else:
        # 제목이 없는 경우 첫 번째 텍스트 박스 내용 추출
        for shape in slide.shapes:
            if hasattr(shape, 'text') and shape.text.strip():
                title = shape.text.split('\n')[0][:30] + "..."
                break

    slide_data.append({"슬라이드 번호": i + 1, "주요 제목": title})

df_overview = pd.DataFrame(slide_data)
display(df_overview)

,슬라이드 번호,주요 제목
0,1,Autonomous Financial Security ...
1,2,The Mechanism of Self-Evolutio...
2,3,Legacy Systems...
3,4,Self-Reflection Cycle...
4,5,Frictionless Oversight...
5,6,The Power of Internal Criticis...
6,7,Long-term Memory...
7,8,99.8%...
8,9,Intelligence increases exponen...
9,10,Phase 1...


In [37]:
from pptx import Presentation
from pptx.util import Inches
import os

# 작업할 파일 경로
ppt_path = '/content/Autonomous_Financial_Security_Center_Styled.pptx'
# 여기에 업로드한 이미지 파일명을 입력하세요 (예: 'my_logo.png')
image_path = '/content/uploaded_image_here.png'

if os.path.exists(ppt_path) and os.path.exists(image_path):
    prs = Presentation(ppt_path)

    # 슬라이드 14 (인덱스 13) 선택
    slide = prs.slides[13]

    # 이미지 위치 및 크기 설정
    left, top, width = Inches(2), Inches(2), Inches(5)
    slide.shapes.add_picture(image_path, left, top, width=width)

    # 저장
    output_name = '/content/Autonomous_Financial_Security_Center_Final_Image.pptx'
    prs.save(output_name)
    print(f"이미지가 성공적으로 삽입되었습니다: {output_name}")
else:
    if not os.path.exists(image_path):
        print(f"오류: '{image_path}' 파일을 찾을 수 없습니다. 이미지 파일을 업로드해 주세요.")
    else:
        print(f"오류: '{ppt_path}' 파일을 찾을 수 없습니다.")

오류: '/content/uploaded_image_here.png' 파일을 찾을 수 없습니다. 이미지 파일을 업로드해 주세요.


In [38]:
import os
import matplotlib.pyplot as plt
from pptx import Presentation
from pptx.util import Inches

# 1. 시각화 이미지 생성 (보안 및 자율 학습 컨셉)
plt.figure(figsize=(6, 4))
labels = ['Autonomous Detection', 'Manual Review', 'Prevented Attacks']
sizes = [75, 5, 20]
colors = ['#003366', '#808080', '#28a745']
plt.pie(sizes, labels=labels, autopct='%1.1f%%', startangle=140, colors=colors)
plt.title('Security Automation Impact v2.0')
img_path = '/content/generated_security_asset.png'
plt.savefig(img_path)
plt.close()

# 2. PPT에 이미지 삽입
ppt_input = '/content/Autonomous_Financial_Security_Center_Refined.pptx'
output_final = '/content/Autonomous_Financial_Security_Center_Final.pptx'

if os.path.exists(ppt_input):
    prs = Presentation(ppt_input)
    slide = prs.slides[13] # 슬라이드 14 (인덱스 13)

    left, top, width = Inches(2), Inches(2.2), Inches(6)
    slide.shapes.add_picture(img_path, left, top, width=width)

    prs.save(output_final)
    print(f'최종 파일 생성 완료: {output_final}')
else:
    print('수정할 기본 PPT 파일을 찾을 수 없습니다.')

최종 파일 생성 완료: /content/Autonomous_Financial_Security_Center_Final.pptx


### 1. 슬라이드 제목 수정 및 순서 변경
아래 코드는 특정 슬라이드의 제목을 수정하고, 슬라이드 순서를 사용자가 원하는 대로 재배치하는 예시입니다.

In [39]:
from pptx import Presentation
import os

def move_slide(prs, old_index, new_index):
    """슬라이드 순서를 변경하는 헬퍼 함수"""
    xml_slides = prs.slides._sldIdLst
    slides = list(xml_slides)
    xml_slides.remove(slides[old_index])
    xml_slides.insert(new_index, slides[old_index])

# 실제 파일 존재 여부 확인 후 경로 설정
input_path = '/content/Autonomous Financial Security Center.pptx'
output_path = '/content/Autonomous_Financial_Security_Center_Final_v2.pptx'

if os.path.exists(input_path):
    prs = Presentation(input_path)

    # 1. 슬라이드 제목 수정 (슬라이드 14, 인덱스 13)
    if len(prs.slides) > 13:
        target_slide = prs.slides[13]
        if target_slide.shapes.title:
            target_slide.shapes.title.text = "Security Performance Visualization"
        else:
            # 제목 개체가 없는 경우 텍스트 박스 추가 시도
            print("슬라이드 14에 제목 개체가 없어 제목을 직접 수정할 수 없습니다.")

    # 2. 슬라이드 순서 변경 예시 (필요시 주석 해제)
    # move_slide(prs, 15, 12)

    prs.save(output_path)
    print(f"슬라이드 수정 및 저장 완료: {output_path}")
else:
    print(f"오류: {input_path} 파일을 찾을 수 없습니다. 파일명을 확인해 주세요.")

오류: /content/Autonomous Financial Security Center.pptx 파일을 찾을 수 없습니다. 파일명을 확인해 주세요.


In [40]:
from pptx import Presentation
import os

# 16개의 슬라이드가 포함된 스타일 적용 버전 사용
input_path = '/content/Autonomous_Financial_Security_Center_Styled.pptx'
output_v2_path = '/content/Autonomous_Financial_Security_Center_Final_v2.pptx'

if os.path.exists(input_path):
    prs = Presentation(input_path)

    # 슬라이드 14 (인덱스 13) 제목 수정
    if len(prs.slides) >= 14:
        slide_14 = prs.slides[13]
        if slide_14.shapes.title:
            slide_14.shapes.title.text = "Security Performance Visualization"
        else:
            # 제목 틀이 없는 경우 새로 생성
            left, top, width, height = Inches(0.5), Inches(0.3), Inches(9), Inches(1)
            title_shape = slide_14.shapes.add_textbox(left, top, width, height)
            title_shape.text = "Security Performance Visualization"

        prs.save(output_v2_path)
        print(f"파일 수정 및 저장 완료: {output_v2_path}")

        # 검증
        check_prs = Presentation(output_v2_path)
        print(f"검증 결과 - 슬라이드 14 제목: {check_prs.slides[13].shapes.title.text}")
    else:
        print(f"오류: 해당 파일의 슬라이드 개수가 {len(prs.slides)}개로 14개 미만입니다.")
else:
    print(f"오류: {input_path} 파일을 찾을 수 없습니다.")

파일 수정 및 저장 완료: /content/Autonomous_Financial_Security_Center_Final_v2.pptx
검증 결과 - 슬라이드 14 제목: Security Performance Visualization


In [41]:
from pptx import Presentation
from IPython.display import display, Markdown
import os

# 수정 대상 및 결과 경로 설정
input_path = '/content/Autonomous_Financial_Security_Center_Styled.pptx'
output_path = '/content/Autonomous_Financial_Security_Center_Final_v2.pptx'

if os.path.exists(input_path):
    prs = Presentation(input_path)
    # 슬라이드 14 (인덱스 13) 제목 수정
    if len(prs.slides) >= 14:
        slide_14 = prs.slides[13]
        if slide_14.shapes.title:
            slide_14.shapes.title.text = "Security Performance Visualization"
        else:
            # 제목 틀이 없는 경우 새로 생성
            from pptx.util import Inches
            left, top, width, height = Inches(0.5), Inches(0.3), Inches(9), Inches(1)
            title_shape = slide_14.shapes.add_textbox(left, top, width, height)
            title_shape.text = "Security Performance Visualization"

        prs.save(output_path)

        # Markdown 형식으로 결과 출력
        display(Markdown(f"### Slide 14 Verification\n**Title:** {slide_14.shapes.title.text if slide_14.shapes.title else 'Security Performance Visualization'}"))
    else:
        print(f"슬라이드 개수 부족: {len(prs.slides)}개")
else:
    print(f"입력 파일을 찾을 수 없습니다: {input_path}")

### Slide 14 Verification
**Title:** Security Performance Visualization

In [42]:
from google.colab import files
import os

file_path = '/content/Autonomous_Financial_Security_Center_Final_v2.pptx'

if os.path.exists(file_path):
    files.download(file_path)
    print(f"Downloading {file_path}...")
else:
    print(f"Error: The file '{file_path}' was not found in the /content/ directory.")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [43]:
import shutil
from google.colab import drive
import os

# Mount Google Drive
drive.mount('/content/drive')

# Define paths
source_path = '/content/Autonomous_Financial_Security_Center_Final_v2.pptx'
destination_path = '/content/drive/MyDrive/Autonomous_Financial_Security_Center_Final_v2.pptx'

# Copy the file
if os.path.exists(source_path):
    shutil.copy(source_path, destination_path)
    print(f"Success! The file has been saved to your Google Drive at: {destination_path}")
else:
    print(f"Error: The source file '{source_path}' does not exist.")

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
Success! The file has been saved to your Google Drive at: /content/drive/MyDrive/Autonomous_Financial_Security_Center_Final_v2.pptx


In [44]:
from pptx import Presentation
import os

file_path = '/content/Autonomous_Financial_Security_Center_Final_v2.pptx'

if os.path.exists(file_path):
    prs = Presentation(file_path)
    print(f"=== 프레젠테이션 요약 (총 {len(prs.slides)}개 슬라이드) ===\n")

    for i, slide in enumerate(prs.slides):
        title = "제목 없음"
        if slide.shapes.title and slide.shapes.title.text.strip():
            title = slide.shapes.title.text.strip()
        else:
            # 제목 개체가 없으면 첫 번째 텍스트 내용 탐색
            for shape in slide.shapes:
                if hasattr(shape, 'text') and shape.text.strip():
                    title = shape.text.split('\n')[0][:50] + "..."
                    break

        print(f"Slide {i+1:02d}: {title}")
else:
    print(f"오류: {file_path} 파일을 찾을 수 없습니다.")

=== 프레젠테이션 요약 (총 20개 슬라이드) ===

Slide 01: Autonomous FinancialSecurity Center...
Slide 02: The Mechanism ofSelf-Evolution...
Slide 03: Legacy Systems...
Slide 04: Self-Reflection Cycle...
Slide 05: Frictionless Oversight...
Slide 06: The Power of Internal Criticism...
Slide 07: Long-term Memory...
Slide 08: 99.8%...
Slide 09: Intelligence increases exponentially as the "Criti...
Slide 10: Phase 1...
Slide 11: Collaborative Agent Roles...
Slide 12: Autonomous Financial Security Center...
Slide 13: Executive Summary
Slide 14: Security Performance Visualization
Slide 15: Threat Detection Metrics
Slide 16: Real-time Security Threat Statistics
Slide 17: Executive Summary
Slide 18: Visualizing Security Data (Image Example)
Slide 19: Threat Detection Metrics
Slide 20: Real-time Security Threat Statistics


In [45]:
from pptx import Presentation
import os

file_path = '/content/Autonomous_Financial_Security_Center_Final_v2.pptx'

if os.path.exists(file_path):
    prs = Presentation(file_path)
    script_context = []

    print("--- 슬라이드 내용 분석 및 대본 생성 준비 ---\n")
    for i, slide in enumerate(prs.slides):
        content = []
        for shape in slide.shapes:
            if hasattr(shape, 'text') and shape.text.strip():
                content.append(shape.text.strip())

        slide_text = " / ".join(content)
        script_context.append(f"Slide {i+1}: {slide_text}")
        print(f"Slide {i+1} 분석 완료")

    # 분석된 내용을 바탕으로 대본 초안 작성 (Markdown)
    print("\n--- 분석 완료! 이제 이 내용을 바탕으로 대본을 작성합니다. ---")
else:
    print(f"오류: {file_path} 파일을 찾을 수 없습니다.")

--- 슬라이드 내용 분석 및 대본 생성 준비 ---

Slide 1 분석 완료
Slide 2 분석 완료
Slide 3 분석 완료
Slide 4 분석 완료
Slide 5 분석 완료
Slide 6 분석 완료
Slide 7 분석 완료
Slide 8 분석 완료
Slide 9 분석 완료
Slide 10 분석 완료
Slide 11 분석 완료
Slide 12 분석 완료
Slide 13 분석 완료
Slide 14 분석 완료
Slide 15 분석 완료
Slide 16 분석 완료
Slide 17 분석 완료
Slide 18 분석 완료
Slide 19 분석 완료
Slide 20 분석 완료

--- 분석 완료! 이제 이 내용을 바탕으로 대본을 작성합니다. ---


### 🎙️ 자율 금융 보안 센터(Autonomous Financial Security Center) 발표 대본

**[Slide 1: 오프닝]**
"안녕하십니까. 오늘 발표할 주제는 금융 보안의 패러다임을 바꿀 '차세대 자율 진화 AI 에이전트 워크플로우'입니다. 수동적인 대응을 넘어, 스스로 학습하고 진화하는 보안 시스템의 미래를 소개해 드리겠습니다."

**[Slide 2-4: 핵심 메커니즘 & 셀프 리플렉션]**
"저희 시스템의 핵심은 '자율 진화'입니다. 단순한 반복 학습이 아니라, 매번 분석이 끝날 때마다 '비판 에이전트(Critic Agent)'가 개입하여 논리적 허점을 찾고 이를 장기 기억 장치에 저장합니다. 이는 다음 공격을 방어할 때 강력한 자산이 됩니다."

**[Slide 11: 협업 에이전트 역할]**
"이 과정은 4가지 핵심 에이전트의 협업으로 이루어집니다. 실시간 탐지를 수행하는 분석가, 이를 검증하는 비판가, 지식을 저장하는 메모리 매니저, 그리고 즉각적인 대응을 실행하는 대응 에이전트가 하나의 유기체처럼 움직입니다."

**[Slide 13: Executive Summary]**
"요약하자면, LLM과 보안 운영의 시너지를 통해 대응 시간(MTTR)을 획기적으로 단축하고, 피드백 기반의 지속적인 진화를 실현하는 것이 본 프로젝트의 목표입니다."

**[Slide 14-16: 데이터 및 성과]**
"보시는 시각화 자료와 같이, 자동화된 탐지 비율이 75%를 상회하며 위협 탐지 정확도는 99.8%에 달합니다. 실시간 위협 통계 데이터를 통해 입증된 것처럼, 저희 시스템은 지금 이 순간에도 진화하고 있습니다."

**[Slide 12: 클로징]**
"진화는 멈추지 않습니다. 자율 AI와 함께 더 안전한 금융의 미래를 준비하십시오. 경청해 주셔서 감사합니다."

In [46]:
from google.colab import files
if os.path.exists('/content/Autonomous_Financial_Security_Center_Final_v2.pptx'):
    files.download('/content/Autonomous_Financial_Security_Center_Final_v2.pptx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [47]:
from google.colab import files
if os.path.exists('/content/Autonomous_Financial_Security_Center_Final.pptx'):
    files.download('/content/Autonomous_Financial_Security_Center_Final.pptx')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [48]:
from pptx import Presentation

# 확인할 파일 경로 설정
file_path = '/content/Autonomous_Financial_Security_Center_Final.pptx'

try:
    prs = Presentation(file_path)
    print(f"--- 프레젠테이션 분석: {file_path} ---")
    print(f"총 슬라이드 수: {len(prs.slides)}\n")

    for i, slide in enumerate(prs.slides):
        print(f"[슬라이드 {i+1}]")

        # 슬라이드 내의 모든 도형(shape)에서 텍스트 추출
        for shape in slide.shapes:
            if hasattr(shape, 'text') and shape.text.strip():
                print(f" - {shape.text.strip()}")
        print("-" * 50)

except Exception as e:
    print(f"파일을 읽는 중 오류가 발생했습니다: {e}")

--- 프레젠테이션 분석: /content/Autonomous_Financial_Security_Center_Final.pptx ---
총 슬라이드 수: 20

[슬라이드 1]
 - Autonomous FinancialSecurity Center
 - Next-Generation Evolving AI Agent Workflows (v2.0)
--------------------------------------------------
[슬라이드 2]
 - The Mechanism ofSelf-Evolution
 - Autonomous learning through iterative feedback loops, transforming raw threat data into actionable security intelligence without manual tuning.
--------------------------------------------------
[슬라이드 3]
 - Legacy Systems
 - Evolving Engine v2
 - •
 - Fixed rule-based detection
 - •
 - Manual updates for new threats
 - •
 - High rate of false positives over time
 - •
 - Zero learning from past cycles
 - •
 - Dynamic agentic workflows
 - •
 - Self-reflection and critic-led feedback
 - •
 - Automatic insight accumulation in Memory
 - •
 - Decreasing error rates via self-correction
 - Static Automation vs. Evolving Intelligence
--------------------------------------------------
[슬라이드 4]
 - Self-Reflecti